### ISSUE TITLE
Coupled Evolution of Deformation, Pore Fluid Pressure, and Fluid Flow in the thrust-fold mountain belt ？

### ISSUE THREAD

**Issue by ch9411** [underworldcode/underworld2#678]
Is it possible to use underworld to build numerical sandbox modeling and research on the coupled Evolution of Deformation, Pore Fluid Pressure, and Fluid Flow in the thrust-fold mountain belt ？

**Comment by bknight1** [comment]
Hi @ch9411, it is possible. We have the stokes flow for the geodynamics, advdiff for temperature evolution and a Darcy solver for fluid flow.

**Comment by ch9411** [comment]
> Hi @ch9411, it is possible. We have the stokes flow for the geodynamics, advdiff for temperature evolution and a Darcy solver for fluid flow.
@bknight1 
Thank you for your response. Due to the limited information available on groundwater flow, especially within the context of UW, we would like to inquire about the setup of permeability and fluid pressure boundaries when using Stokes flow. Can the external fluid pressure vary with deformation (in other words, coupling)? We look forward to your reply. Thank you.

**Comment by bknight1** [comment]
Yes, the two can be coupled. You can set up each material to have a specific permeability or you can have a mesh/swarm variable that describes the permeability in the model that is used to calculate the fluid flow. The pressure field can be obtained from the Stokes flow and then used in the Darcy solve to determine the groundwater flow.

**Comment by ch9411** [comment]
> Yes, the two can be coupled. You can set up each material to have a specific permeability or you can have a mesh/swarm variable that describes the permeability in the model that is used to calculate the fluid flow. The pressure field can be obtained from the Stokes flow and then used in the Darcy solve to determine the groundwater flow.

@bknight1 
Your response has enlightened me, but as I am a beginner, I am not very familiar with the mechanisms involved. Could you please provide a simple example or code? Thank you very much!
